saleor/graphql/product/tests/test_product.py

```diff
@@ -1786,27 +1837,70 @@ def test_products_query_with_filter_search_by_sku(
     assert products[0]["node"]["name"] == product_with_default_variant.name
 
 
-def test_products_query_with_filter_stock_availability(
+def test_products_query_with_filter_stock_availability_as_staff(
     query_products_with_filter,
     staff_api_client,
-    product,
+    product_list,
     order_line,
     permission_manage_products,
+    channel_USD,
 ):
-    stock = product.variants.first().stocks.first()
-    Allocation.objects.create(
-        order_line=order_line, stock=stock, quantity_allocated=stock.quantity
+
+    for product in product_list:
+        stock = product.variants.first().stocks.first()
+        Allocation.objects.create(
+            order_line=order_line, stock=stock, quantity_allocated=stock.quantity
+        )
+    product = product_list[0]
+    product.variants.first().channel_listings.filter(channel=channel_USD).update(
+        price_amount=None
     )
-    variables = {"filter": {"stockAvailability": "OUT_OF_STOCK"}}
+
+    variables = {
+        "filter": {"stockAvailability": "OUT_OF_STOCK", "channel": channel_USD.slug}
+    }
     staff_api_client.user.user_permissions.add(permission_manage_products)
     response = staff_api_client.post_graphql(query_products_with_filter, variables)
     content = get_graphql_content(response)
-    product_id = graphene.Node.to_global_id("Product", product.id)
     products = content["data"]["products"]["edges"]
 
-    assert len(products) == 1
+    assert len(products) == 3
+
```

Comment:
Same here.

Reply:
In this case we are receiving all products with and without price.